pkgs/top-level/all-packages.nix

```diff
@@ -24069,31 +24069,7 @@ with pkgs;
 
   nvitop = callPackage ../tools/system/nvitop { };
 
-  nvtop = callPackage ../tools/system/nvtop { };
-  nvtop-amd = (callPackage ../tools/system/nvtop {
-    amd = true;
-    intel = false;
-    msm = false;
-    nvidia = false;
-  }).overrideAttrs { pname = "nvtop-amd"; };
-  nvtop-intel = (callPackage ../tools/system/nvtop {
-    amd = false;
-    intel = true;
-    msm = false;
-    nvidia = false;
-  }).overrideAttrs { pname = "nvtop-intel"; };
-  nvtop-msm = (callPackage ../tools/system/nvtop {
-    amd = false;
-    intel = false;
-    msm = true;
-    nvidia = false;
-  }).overrideAttrs { pname = "nvtop-msm"; };
-  nvtop-nvidia = (callPackage ../tools/system/nvtop {
-    amd = false;
-    intel = false;
-    msm = false;
-    nvidia = true;
-  }).overrideAttrs { pname = "nvtop-nvidia"; };
+  nvtopPackages = import ../tools/system/nvtop { inherit callPackage; };
```

Comment:
You can still use `callPackage`, and then you won't have to `inherit callPackage`.

Reply:
Oh on a 2nd thought, you should use here `recureseIntoAttrs` somehow, so Hydra will build all of these attributes.

Reply:
When I run `nix-build -A nvtopPackages` locally, it does build all 7 packages. Maybe `recureseIntoAttrs` is not necessary?

Reply:
> When I run `nix-build -A nvtopPackages` locally, it does build all 7 packages. Maybe `recureseIntoAttrs` is not necessary?

The fact you can access this attribute via `nix-build` or `nix build` doesn't mean Hydra will collect it when it'll search for packages to build. You can test whether these attributes are built by using `nix search . nvtop`.

Reply:
TBH specifically for these new attributes, since many of them require non-free dependencies, Hydra won't build them too, but for a different reason :). Still though, making them appear in `nix search nixpkgs` is important.

Reply:
Added recurseIntoAttrs and validated that nix search . nvtop works.